feature-libs/product/multi-dimensional/components/multi-dimensional-container/multi-dimensional-selector/product-multi-dimensional-selector.component.ts

```diff
@@ -61,7 +68,50 @@ export class ProductMultiDimensionalSelectorComponent implements OnChanges {
     }
   }
 
-  isSelected(code: string) {
+  getSelectedValue(categoryName: string): string | null {
+    const category = this.categories.find((cat) => cat.name === categoryName);
+
+    if (category && this.selectedVariant) {
+      const selectedOption = category.variantOptions.find(
+        (option) => option.code === this.selectedVariant
+      );
+      return selectedOption ? selectedOption.value : '';
+    }
+
+    return null;
+  }
+
+  isSelected(code: string): boolean {
     return code === this.product.code;
   }
+
+  getAriaLabel(option: VariantCategoryOption, categoryName: string) {
```

Comment:
In general, we should avoid subscribing directly in the template using methods. Because subscriptions are asynchronous, this practice can lead to hard-to-find bugs.

```typescript
 onAriaLabel(option: VariantCategoryOption, categoryName: string): Observable<string> {
    const isSelected = this.isSelected(option.code);

    if (isSelected) {
      return this.translationService
        .translate('multiDimensional.selectedVariant')
        .pipe(map(
          (text) =>
            `${text}, ${option.value} ${categoryName}`
        ));
    }

    return this.translationService
      .translate('multiDimensional.variantThumbnailTitle', {
        value: option.value,
        category: categoryName,
      })
      .pipe(map((text) => `${text}`));
  }
```

Reply:
Changed